mmdet3d/datasets/transforms/transforms_3d.py

```diff
@@ -2352,3 +2355,189 @@ def transform(self, input_dict: dict) -> dict:
+            # calculate horizontal angle for each point
+            yaw = -torch.atan2(points.coord[:, 1], points.coord[:, 0])
+            retrieve_yaw = -torch.atan2(retrieve_points.coord[:, 1],
+                                        retrieve_points.coord[:, 0])
+
+            # select points in sector
+            idx = (yaw <= start_angle) | (yaw >= end_angle)
+            retrieve_idx = (retrieve_yaw > start_angle) & (
+                retrieve_yaw < end_angle)
+
+            # swap
+            points = points.cat([points[idx], retrieve_points[retrieve_idx]])
+            pts_semantic_mask = np.concatenate(
+                (pts_semantic_mask[idx.numpy()],
+                 retrieve_pts_semantic_mask[retrieve_idx.numpy()]),
+                axis=0)
+
+        # 2. rotate-pasting
+        if np.random.random() < self.rotate_paste_ratio:
+            # extract instance points
+            instance_points, instance_pts_semantic_mask = [], []
+            for instance_class in self.instance_classes:
+                retrieve_idx = retrieve_pts_semantic_mask == instance_class
+                instance_points.append(retrieve_points[retrieve_idx])
+                instance_pts_semantic_mask.append(
+                    retrieve_pts_semantic_mask[retrieve_idx])
+            instance_points = retrieve_points.cat(instance_points)
+            instance_pts_semantic_mask = np.concatenate(
+                instance_pts_semantic_mask, axis=0)
+
+            # rotate-copy
+            copy_points = [instance_points]
+            copy_pts_semantic_mask = [instance_pts_semantic_mask]
+            angle_list = [
+                np.random.random() * np.pi * 2 / 3,
+                (np.random.random() + 1) * np.pi * 2 / 3
+            ]
+            for angle in angle_list:
+                new_points = instance_points.clone()
+                new_points.rotate(angle)
+                copy_points.append(new_points)
+                copy_pts_semantic_mask.append(instance_pts_semantic_mask)
+            copy_points = instance_points.cat(copy_points)
+            copy_pts_semantic_mask = np.concatenate(
+                copy_pts_semantic_mask, axis=0)
+
+            points = points.cat([points, copy_points])
+            pts_semantic_mask = np.concatenate(
+                (pts_semantic_mask, copy_pts_semantic_mask), axis=0)
+
+        input_dict['points'] = points
+        input_dict['pts_semantic_mask'] = pts_semantic_mask
+        return input_dict
+
+    def transform(self, input_dict: dict) -> dict:
+        """PolarMix transform function.
+
+        Args:
+            input_dict (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: output dict after transformtaion.
+        """
+
+        assert 'dataset' in input_dict
+        dataset = input_dict.pop('dataset', None)
+
+        # get index of other images
+        index = self.get_indexes(dataset)
+
+        mix_results = [copy.deepcopy(dataset.get_data_info(index))]
```

Comment:
`copy.deepcopy` is unnecessary since the returns of `get_data_info` is a deep copy of the original data info.